tests/test_service_account.py

```diff
@@ -173,6 +173,30 @@ def test_from_p12_keyfile_explicit(self):
         self._from_p12_keyfile_helper(private_key_password=password,
                                       scopes=['foo', 'bar'])
 
+    def test_from_p12_keyfile_buffer(self):
+        service_account_email = 'name@email.com'
+        filename = data_filename('privatekey.p12')
+        private_key_password = 'notasecret'
+        scopes = ['foo', 'bar']
+        with open(filename, 'rb') as file_obj:
+            key_contents = file_obj.read()
+            # Seek back to the beginning so the buffer can be
+            # passed to the constructor.
+            file_obj.seek(0)
+            creds = ServiceAccountCredentials.from_p12_keyfile_buffer(
+                service_account_email, file_obj,
+                private_key_password=private_key_password,
+                scopes=scopes)
+        # Check the created object.
+        self.assertIsInstance(creds, ServiceAccountCredentials)
+        self.assertEqual(creds.client_id, None)
```

Comment:
Since this is already a `unittest2`-using file, `assertIsNone`, please (here and below).